docs/features/kubernetes/troubleshooting.md

```diff
@@ -0,0 +1,103 @@
+---
+id: troubleshooting
+title: Troubleshooting Kubernetes
+sidebar_label: Troubleshooting
+description: Troubleshooting for Kubernetes
+---
+
+## Kubernetes is not showing up on Service Entities
+
+Sometimes, Kubernetes is not showing up on service entities, we should test your
+k8s cluster are already connected to Backstage or not.
+
+```curl
+curl --location --request POST '{{backstage-backend-url}}:{{backstage-backend-port}}/api/kubernetes/services/:service-entity-name' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "entity": {
+        "metadata": {
+            "name": <service-entity-name>
+        }
+    }
+}
+'
+```
+
+The curl response should have resources from Kubernetes:
+
+```json
+# curl response
+{
+  "items": [
+    {
+      "cluster": {
+        "name": <cluster-name>
+      },
+      "resources": [
+        {
+          "type": "services",
+          "resources": [
+            {
+              "metadata": {
+                "creationTimestamp": "2022-03-13T13:52:46.000Z",
+                "labels": {
+                  "app": <k8s-app-name>,
+                  "backstage": <selector>,
+                  "backstage.io/kubernetes-id": <service-entity-name>
+                },
+                "name": <k8s-app-name>,
+                "namespace": <namespace>
+              },
+              ....
+            }
+          ]
+        },
+        ....
+        {
+          "type": "pods",
+          "resources": [
+            ,,,,
+          ]
+        }
+      ],
+      "errors": []
+    }
+  ]
+}
+
+```
+
+Kubernetes will not be showing anything when catalog info annotations is not
+match with k8s related yaml label (service.yaml, deployment.yaml, etc). We
+recommend you for adding two labels and using label selector annotations:
```

Comment:
> The Kubernetes tab will not show anything when the catalog info annotation does not
match the related Kubernetes resource. We recommend you add the following labels to your resources and use the label selector annotation as follows: